src/core/Track.cpp

```diff
@@ -159,14 +159,16 @@ void TrackContentObject::movePosition( const MidiTime & pos )
 
 /*! \brief Change the length of this TrackContentObject
  *
- *  If the track content object's length has chaanged, update it.  We
+ *  If the track content object's length has changed, update it.  We
  *  also add a journal entry for undo and update the display.
  *
  * \param _length The new length of the track content object.
  */
 void TrackContentObject::changeLength( const MidiTime & length )
 {
-	m_length = length;
+	// use a minimum of 1 bar always, otherwise it's possible to get 0 length TCOs
+	// which will make you think it has disappeared
+	m_length = qMax(MidiTime(1, 0), length);
```

Comment:
As mentioned in comment, clips shorter than a bar are valid and shouldn't be prevented.